doc/user/content/transform-data/troubleshooting.md

```diff
@@ -93,11 +95,35 @@ this means that the index was correctly used. If that's not the case, consider:
 * Does the index's indexed expression (key) match up with how you're querying
   the data?
 
-#### Computation-intensive queries
-
-Another thing to be aware of, in Materialize, aggregations like `COUNT(*)` and limit constraints that return more than 25 rows like `LIMIT 50` perform a computation across the full underlying dataset being queried, so can be resource intensive. If you want to execute these types of queries you should either:
-* Create a materialized view or index for these queries if you plan to issue them often
-* Use a large enough cluster to fit the full underlying dataset being queried
+#### Resource-intensive queries
+##### Data-intensive queries
+Queries with aggregations, ordering, or offsets can be data-intensive, because
```

Comment:
I think pretty much all queries can be data intensive. Everything except for persist limit pushdown and some persist filtered data would be "data intensive" when derived from persist data. Queries that draw out of indexed views may also be data intensive, depending on what happens after the index.

I don't think it has much to do with having aggregations, ordering, or offsets. If you have none of those but have a `JOIN` it will still be data-intensive. If you have none of those and literally just type `SELECT * FROM ..` it will be data intensive.

They are also only data intensive if read from persist. Doing a `SELECT * FROM indexed ORDER BY whatever LIMIT many` will only scan the index and maintain `many` records in memory.